aptos-move/e2e-move-tests/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos::common::types::MovePackageDir;
+use aptos::move_tool::{BuiltPackage, MemberId};
+use aptos_types::access_path;
+use aptos_types::account_address::AccountAddress;
+use aptos_types::transaction::{ScriptFunction, TransactionPayload, TransactionStatus};
+use aptos_vm::move_vm_ext::UpgradePolicy;
+use cached_framework_packages::aptos_stdlib;
+use language_e2e_tests::account::{Account, AccountData};
+use language_e2e_tests::executor::FakeExecutor;
+use move_deps::move_core_types::language_storage::{StructTag, TypeTag};
+use project_root::get_project_root;
+use serde::de::DeserializeOwned;
+use std::collections::BTreeMap;
+use std::path::Path;
+
+/// A simple test harness for defining Move e2e tests.
+///
+/// Tests defined via this harness typically live in the `<crate>/tests` directory, the standard
+/// Rust place for defining integration tests.
+///
+/// For defining a set of new tests around a specific area, you add a new Rust source
+/// `tested_area.rs` to the `tests` directory of your crate. You also will create a directory
+/// `tested_area.data` which lives side-by-side with the Rust source. In this directory, you
+/// place any number of Move packages you need for running the tests. In addition, the test
+/// infrastructure will place baseline (golden) files in the `tested_area.data` using the `.exp`
+/// (expected) ending.  For examples, see e.g. the `tests/code_publishing.rs` test in this crate.
+///
+/// NOTE: This harness currently is a wrapper around existing legacy e2e testing infra. We
+/// eventually plan to retire the legacy code, and are rather keen to know what of the legacy
+/// test infra we want to maintain and also which existing tests to preserve.
+pub struct MoveHarness {
+    /// The executor being used.
+    executor: FakeExecutor,
+    /// The current transaction sequence number, by account address.
+    txn_seq_no: BTreeMap<AccountAddress, u64>,
+}
+
+impl MoveHarness {
+    /// Creates a new harness.
+    pub fn new() -> Self {
+        Self {
+            executor: FakeExecutor::from_fresh_genesis(),
+            txn_seq_no: BTreeMap::default(),
+        }
+    }
+
+    /// Creates an account for the given static address. This address needs to be static so
+    /// we can load regular Move code to there without need to rewrite code addresses.
+    pub fn new_account_at(&mut self, addr: AccountAddress) -> Account {
+        // The below will use the genesis keypair but that should be fine.
+        let acc = Account::new_genesis_account(addr);
+        let data = AccountData::with_account(acc, 1_000_000, 10);
+        self.executor.add_account_data(&data);
+        data.account().clone()
+    }
+
+    /// Runs a transaction, based on provided payload. If the transaction succeeds, any generated
+    /// writeset will be applied to storage.
+    pub fn run_txn(&mut self, account: &Account, payload: TransactionPayload) -> TransactionStatus {
```

Comment:
we may want to actually submit transactions to test out gas, sequence numbers, and multiagent, it might make sense to rename this run_transaction_payload that way we don't need to do a codemod when we decide to expose more functionality.

also we're not short on space so prefer transaction over txn